datascience/tables.py

```diff
@@ -1155,12 +1155,54 @@ def stack(self, key, labels=None):
         return type(self)([key, 'column', 'value']).with_rows(rows)
 
     def join(self, column_label, other, other_label=None):
-        """Generate a table with the columns of self and other, containing rows
+        """Generates a new table with the columns of self and other, containing rows
         for all values of a column that appear in both tables.
-        If a join value appears more than once in self, each row will be used,
-        but in the other table, only the first of each will be used.
```

Comment:
I think this comment is an important implementation detail! But right now it's just worded weirdly. Basically, the behavior of `join` is:

```
In [2]: t = Table().with_columns('x', [1, 2, 3], 'y', [10, 11, 12])
In [13]: t2 = Table().with_columns('x', [1, 1, 3], 'z', [100, 105, 110])

In [12]: t
Out[12]:
x    | y
1    | 10
2    | 11
3    | 12
In [14]: t2 # t2 has x = 1 twice
Out[14]:
x    | z
1    | 100
1    | 105
3    | 110

# But when we join, only the first 'x' value in t2 is kept
In [15]: t.join('x', t2)
Out[15]:
x    | y    | z
1    | 10   | 100
3    | 12   | 110

# Same for the opposite join
In [17]: t2.join('x', t)
Out[17]:
x    | z    | y
1    | 100  | 10
1    | 105  | 10
3    | 110  | 12
```

This behavior is a bit strange but it tends to work out for students. I think we should clarify this behavior in the docs and probably add examples demonstrating this.